### PR TITLE
Message sign verify functionality compatible with Bitcoin core & electrum

### PR DESCRIPTION
**Pull request by ghost**
Sign/verify functionality which is comparable to Electrum.  Requested by @huey735 

- Can sign a message using any address in your wallet.
- Can verify any message+address+signature
- Located in the Multisig Payout Tool (CTRL-G)
 

Screenshot: Bisq sign/verify & Electrum sign/verify.

image

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/overlays/windows/ManualPayoutTxWindow.java [hunk 1]
```diff
@@ -481,12 +493,70 @@ private void addBuildPane() {
...
+        myGridPane.add(buttonBox, 0, ++rowIndexB);
+
+        buttonSign.setOnAction(e -> {
+            String walletInfo = walletsManager.getWalletsAsString(true);
+            String privKeyHex = findPrivForPubOrAddress(walletInfo, address.getText());
+            if (privKeyHex == null) {
+                messageSig.setText("");
+                new Popup().information("Key not found in wallet").show();
+            } else {
+                ECKey myPrivateKey = ECKey.fromPrivate(Utils.HEX.decode(privKeyHex));
```
Comment (wallclockbuilder) [inline desktop/src/main/java/bisq/desktop/main/overlays/windows/ManualPayoutTxWindow.java:529]: [Tested](https://github.com/bitcoinj/bitcoinj/blob/68097e11f673ddaf16ec4e2f71f7e676d581f350/core/src/test/java/org/bitcoinj/core/ECKeyTest.java#L210) by BitcoinJ.

File: desktop/src/main/java/bisq/desktop/main/overlays/windows/ManualPayoutTxWindow.java [hunk 2]
```diff
@@ -481,12 +493,70 @@ private void addBuildPane() {
...
+                String signatureBase64 = myPrivateKey.signMessage(messageText.getText());
+                messageSig.setText(signatureBase64);
+            }
+        });
+        buttonVerify.setOnAction(e -> {
+            try {
+                ECKey key = ECKey.signedMessageToKey(messageText.getText(), messageSig.getText());
```
Comment (wallclockbuilder) [inline desktop/src/main/java/bisq/desktop/main/overlays/windows/ManualPayoutTxWindow.java:536]: [Tested](https://github.com/bitcoinj/bitcoinj/blob/68097e11f673ddaf16ec4e2f71f7e676d581f350/core/src/test/java/org/bitcoinj/core/ECKeyTest.java#L226) by Bitcoinj.

**Review by wallclockbuilder — APPROVED** [review]
ACK


### Sign

- Is Bisq compatible with Electrum?


1 1electrum message - address - signature
Message signed by address in Electrum

1 2bisq compatible with electrum
Same message signed by Bisq with the same address resulted in the same signature.
<br/>
<br/>
<br/>



### Verify

- Can Bisq Verify A message signed by Electrum?
<br/>
<br/>


2electrum signed
Message signed by electrum. Will Bisq be able to verify it?
<br/>
<br/>

2 1bisq verification setup
Load signed message details from Electrum into Bisq. Will it verify?
<br/>
<br/>

2 2bisq signature verified
Bisq verifies signature from Electrum successfully
<br/>
<br/>
<br/>


### Unhappy path
3 1bisq - wrong message - setup
Modify message. Will Bisq be fooled?
<br/>
<br/>

3 2bisq wrong signature - message changed
Bisq successfully detects wrong signature due to modified message.
<br/>
<br/>

4 1bisq wrong address setup
Tamper with address. Will bisq be able to tell?
<br/>
<br/>

4 2bisq wrong signature - addess changed
Bisq successfully detects wrong signature due to wrong address.
<br/>
<br/>

5 1bisq wrong signature setup
Tamper with signature. Will bisq be able to tell?
<br/>
<br/>

5 2bisq wrong signature - signature changed
Bisq successfully rejects wrong signature.

**Review by ripcurlx — APPROVED** [review]
utACK

**Pull request merged**